src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java

```diff
@@ -77,19 +82,39 @@ void dectectOnchainCompleted(List<TaskStatus> dectectWhenOffchainTaskStatuses,
         }
     }
 
-    void updateReplicateStatuses(String chainTaskId, Replicate replicate, ReplicateStatus offchainCompleted) {
+    private void updateReplicateStatuses(String chainTaskId, Replicate replicate, ReplicateStatus offchainCompleted) {
         List<ReplicateStatus> statusesToUpdate;
         if (replicate.getCurrentStatus().equals(WORKER_LOST)) {
             statusesToUpdate = getMissingStatuses(replicate.getLastButOneStatus(), offchainCompleted);
         } else {
             statusesToUpdate = getMissingStatuses(replicate.getCurrentStatus(), offchainCompleted);
         }
 
+        String wallet = replicate.getWalletAddress();
+
         for (ReplicateStatus statusToUpdate : statusesToUpdate) {
-            replicatesService.updateReplicateStatus(chainTaskId, replicate.getWalletAddress(),
-                    statusToUpdate, ReplicateStatusModifier.POOL_MANAGER);
-        }
-    }
+            // add details to the update if needed
+            switch (statusToUpdate) {
```

Comment:
What do you think about moving this block to the real updateReplicateStatus(..) so we can remove the blockNumber from the update method (since we can't trust the worker) ?